src/core/server/saved_objects/service/lib/integration_tests/repository_with_proxy.test.ts

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Hapi from '@hapi/hapi';
+import h2o2 from '@hapi/h2o2';
+import { URL } from 'url';
+import { ISavedObjectsRepository } from '../repository';
+import { SavedObject } from '../../../types';
+import { InternalCoreSetup, InternalCoreStart } from '../../../../internal_types';
+import { Root } from '../../../../root';
+import * as kbnTestServer from '../../../../../test_helpers/kbn_server';
+import {
+  declareGetRoute,
+  declareDeleteRoute,
+  declarePostBulkRoute,
+  declarePostMgetRoute,
+  declareGetSearchRoute,
+  declarePostSearchRoute,
+  declarePostUpdateRoute,
+  declarePostPitRoute,
+  declarePostUpdateByQueryRoute,
+  declarePassthroughRoute,
+  setProxyInterrupt,
+  getProxyInterrupt,
+} from './repository_with_proxy_utils';
+
+let esServer: kbnTestServer.TestElasticsearchUtils;
+const registerSOTypes = (setup: InternalCoreSetup) => {
+  setup.savedObjects.registerType({
+    name: 'my_type',
+    hidden: false,
+    mappings: {
+      dynamic: false,
+      properties: {
+        title: { type: 'text' },
+      },
+    },
+    namespaceType: 'single',
+  });
+  setup.savedObjects.registerType({
+    name: 'my_other_type',
+    hidden: false,
+    mappings: {
+      dynamic: false,
+      properties: {
+        title: { type: 'text' },
+      },
+    },
+    namespaceType: 'single',
+  });
+};
+// let proxyInterrupt: string | null | undefined = null;
+
+describe('404s from proxies', () => {
+  let root: Root;
+  let start: InternalCoreStart;
+  let hapiServer: Hapi.Server;
+
+  beforeAll(async () => {
+    setProxyInterrupt(null);
+
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+    });
+    esServer = await startES();
+
+    const esUrl = new URL(esServer.hosts[0]);
+    // For the proxy, use a port number that is 100 higher than the one that the actual ES instance is using
+
+    const proxyPort = parseInt(esUrl.port, 10) + 100;
```

Comment:
> use a port number that is 100 higher than the one that the actual ES instance is using

I'm not sure you can rely on this being the case. In the CI configs, ES runs on 6201-6299 ports, and ES transport on 6301-6399. So you are probably running into conflicts with the transport ports here: 

https://github.com/elastic/kibana/blob/d2690d8ac81622aa0c72c4fe17d3f243427628ee/vars/kibanaPipeline.groovy#L83-L94 

This is probably why you are seeing failures on CI but not locally. You will probably need to use an environment variable to configure this port, and then pass it in from the groovy config. For example, you could probably use:
```groovy
  def proxyTestServerPort = "64${parallelId}4"
```

I ran into something similar in https://github.com/elastic/kibana/pull/100957, which has an example of how you can wire this up. Feel free to give me a ping if you want to talk it through!